doc/tutorials/lattice_boltzmann/lattice_boltzmann_sedimentation.ipynb

```diff
@@ -0,0 +1,396 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.animation as animation\n",
+    "import tempfile\n",
+    "import base64\n",
+    "\n",
+    "from matplotlib.quiver import Quiver\n",
+    "\n",
+    "VIDEO_TAG = \"\"\"<video controls>\n",
+    " <source src=\"data:video/x-m4v;base64,{0}\" type=\"video/mp4\">\n",
+    " Your browser does not support the video tag.\n",
+    "</video>\"\"\"\n",
+    "\n",
+    "\n",
+    "def anim_to_html(anim):\n",
+    "    if not hasattr(anim, '_encoded_video'):\n",
+    "        with tempfile.NamedTemporaryFile(suffix='.mp4') as f:\n",
+    "            anim.save(f.name, fps=20, extra_args=['-vcodec', 'libx264'])\n",
+    "            with open(f.name, \"rb\") as g:\n",
+    "                video = g.read()\n",
+    "        anim._encoded_video = base64.b64encode(video).decode('ascii')\n",
+    "        plt.close(anim._fig)\n",
+    "    return VIDEO_TAG.format(anim._encoded_video)\n",
+    "\n",
+    "\n",
+    "animation.Animation._repr_html_ = anim_to_html\n",
+    "\n",
+    "# set ignore 'divide' and 'invalid' errors\n",
+    "# these occur when plotting the flowfield containing a zero velocity\n",
+    "np.seterr(divide='ignore', invalid='ignore')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And now the actual visualization code."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# setup figure and prepare axes\n",
+    "fig = plt.figure(figsize=(2 * 5, 5 / box_width * box_height))\n",
+    "gs = fig.add_gridspec(1, 2, wspace=0.1)\n",
+    "(ax1, ax2) = gs.subplots(sharey=True)\n",
+    "\n",
+    "ax1.set_title(\"Langevin\")\n",
+    "ax1.set_xlim((0, box_width))\n",
+    "ax1.set_ylim((0, box_height))\n",
+    "\n",
+    "ax2.set_title(\"LB Fluid\")\n",
+    "ax2.set_xlim((0, box_width))\n",
+    "ax2.set_ylim((0, box_height))\n",
+    "\n",
+    "# draw walls\n",
+    "for ax in [ax1, ax2]:\n",
+    "    ax.hlines((1, box_height-1), 0, box_width, color=\"gray\")\n",
+    "\n",
+    "# create meshgrid for quiver plot\n",
+    "xs = np.array([x for x in range(n_width)]) * spacing\n",
+    "ys = np.array([y for y in range(n_height)]) * spacing\n",
+    "X, Y = np.meshgrid(xs, ys)\n",
+    "\n",
+    "# initialize plot objects\n",
+    "lb_ff = ax2.quiver(X, Y, data_flowfield[0, :, :, 0],\n",
+    "                   data_flowfield[0, :, :, 1])\n",
+    "lb_particles, = ax2.plot([], [], 'o')\n",
+    "lv_particles, = ax1.plot([], [], 'o')\n",
+    "\n",
+    "def draw_frame(t):\n",
+    "    # manually remove Quivers from ax2\n",
+    "    for artist in ax2.get_children():\n",
+    "        if isinstance(artist, Quiver):\n",
+    "            artist.remove()\n",
+    "    \n",
+    "    # draw new quivers\n",
+    "    lb_ff = ax2.quiver(X, Y, data_flowfield[t, :, :, 0],\n",
+    "                       data_flowfield[t, :, :, 1])\n",
```

Comment:
By default, matplotlib quivers are automatically scaled. As a result, at the beginning of the simulation when the LB velocity is in the range $10^{-12}$, the arrows flip up and down to adjust to the random noise, which causes the screen to flicker. Also, the fluid velocity inside the boundary is used to create arrows that give the impression the top LB boundary creates a fluid flow. Please find below a solution that rescales arrow lengths based on the largest velocity in the entire trajectory and removes the top and bottom rows of the LB fluid.

```suggestion
    "xs = np.array([x for x in range(n_width)]) * spacing\n",
    "ys = np.array([y for y in range(1, n_height-1)]) * spacing\n",
    "X, Y = np.meshgrid(xs, ys)\n",
    "\n",
    "# set quiver scale (fraction of the highest velocity in the XY plane)\n",
    "lb_vel_max = np.sum(np.square(data_flowfield[:, :, :, 0:-1]), axis=-1)\n",
    "quiver_scale = np.max(lb_vel_max) / 4.\n",
    "\n",
    "def plot_lb_vel(ax, X, Y, data_flowfield, t, scale):\n",
    "    # skip data inside the LB boundaries (top and bottom walls)\n",
    "    return ax.quiver(X, Y,\n",
    "                     data_flowfield[t, 1:-1, :, 0],\n",
    "                     data_flowfield[t, 1:-1, :, 1],\n",
    "                     scale_units=\"xy\", scale=scale)\n",
    "\n",
    "# initialize plot objects\n",
    "lb_ff = plot_lb_vel(ax2, X, Y, data_flowfield, 0, quiver_scale)\n",
    "lb_particles, = ax2.plot([], [], 'o')\n",
    "lv_particles, = ax1.plot([], [], 'o')\n",
    "\n",
    "def draw_frame(t):\n",
    "    # manually remove Quivers from ax2\n",
    "    for artist in ax2.get_children():\n",
    "        if isinstance(artist, Quiver):\n",
    "            artist.remove()\n",
    "    \n",
    "    # draw new quivers\n",
    "    lb_ff = plot_lb_vel(ax2, X, Y, data_flowfield, t, quiver_scale)\n",
```